pkg/store/cache/inmemory.go

```diff
@@ -269,15 +271,29 @@ func (c *InMemoryIndexCache) reset() {
 	c.curSize = 0
 }
 
+func copyString(s string) string {
+	var b []byte
+	h := (*reflect.SliceHeader)(unsafe.Pointer(&b))
+	h.Data = (*reflect.StringHeader)(unsafe.Pointer(&s)).Data
+	h.Len = len(s)
+	h.Cap = len(s)
+	return string(b)
+}
+
+// copyToKey is required as underlying strings might be mmaped.
+func copyToKey(l labels.Label) cacheKeyPostings {
```

Comment:
I think this should probably be named simply `copyCacheKey` or `copyKey` for better readability